### PR TITLE
Add `Makefile.win`

### PR DESCRIPTION
**Pull request by HertzDevil**
The docs-related targets are dropped, because manfiles do not make sense on Windows. Everything else should work, in particular setting up the Git commit hash and the build date for `bin\shards.exe --version`. Setting `%SHARDS%` to `false` also works as long as both `curl` and `tar` are available ([Windows 10 should come with them already](https://learn.microsoft.com/en-us/virtualization/community/team-blog/2017/20171219-tar-and-curl-come-to-windows)).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -60,10 +60,10 @@ jobs:
         uses: actions/checkout@v2
 
       - name: Build
-        run: make
+        run: make -f ${{ runner.os == 'Windows' && 'Makefile.win' || 'Makefile' }}
```
Comment (Sija) [inline .github/workflows/ci.yml:63]: I'd suggest using ternary operator as it's more straightforward.

```suggestion
        run: make -f ${{ runner.os == 'Windows' ? 'Makefile.win' : 'Makefile' }}
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -60,10 +60,10 @@ jobs:
...
 
       - name: Run specs
-        run: make test
+        run: make -f ${{ runner.os == 'Windows' && 'Makefile.win' || 'Makefile' }} test
```
Comment (Sija) [inline .github/workflows/ci.yml:66]: ditto

```suggestion
        run: make -f ${{ runner.os == 'Windows' ? 'Makefile.win' : 'Makefile' }} test
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (HertzDevil) [reply .github/workflows/ci.yml:63]: [They are not available](https://github.com/orgs/community/discussions/75928)

Comment (Sija) [reply .github/workflows/ci.yml:63]: Wow, GHA weirdness make my head spin sometimes 🤯 Thanks for the clarification.

**Review by straight-shoota — APPROVED** [review]
Sigh, I don't like this duplication at all. 😢  But I guess there's no other way (without throwing everything over).

**Pull request merged**